### PR TITLE
Functions: test keyword parameter

### LINKED ISSUE #221: Functions: test positional vs keyword arguments
Positional vs keyword arguments (use the [`inspect`](https://docs.python.org/3/library/inspect.html#inspect.Parameter) module for testing)

_Originally posted by @edoardob90 in https://github.com/empa-scientific-it/python-tutorial/issues/118#issuecomment-2007242461_

### PR DESCRIPTION
**Pull request by despadam**
Fixes #221

### CODE REVIEW

**Review by baffelli — CHANGES_REQUESTED** [review]
We should consider how to deal with multiple asserts. 
Only the first assertion to fail is shown in the results. This is a design limitation of pytest

**Review by baffelli — DISMISSED** [review]
The representation of the error is not ideal, but I suspect there's no better easy way to solve this.

File: tutorial/tests/test_functions.py [hunk 1]
```diff
@@ -22,34 +22,46 @@ def reference_greet(name: str, age: int) -> str:
...
+        errors.append("The function's return value is missing the type hint.")
 
-    assert all(
-        p.annotation != inspect.Parameter.empty for p in params.values()
-    ), "The function's parameters should have type hints"
-    assert (
-        return_annotation != inspect.Signature.empty
-    ), "The function's return value is missing the type hint"
+    # test signature
+    assert not errors
```
[the remark this answers is not in the document]
Comment (despadam) [reply tutorial/tests/test_functions.py]: The latest commit prints the errors in this way:
Capture

[the remark this answers is not in the document]
Comment (baffelli) [reply tutorial/tests/test_functions.py]: I like this a lot!
Can you add a bullet list and a title?
Thank you again!

[the remark this answers is not in the document]
Comment (despadam) [reply tutorial/tests/test_functions.py]: image

**Review by baffelli — COMMENTED** [review]
Perfect. Only suggestion would be to avoid displaying the assertion below the list, but I'm not sure whether this is possible.

**Comment by despadam** [comment]
> Perfect. Only suggestion would be to avoid displaying the assertion below the list, but I'm not sure whether this is possible.

To do that, we would have to change how the test output is displayed in general, but I do not want to touch that now. I think in general testsuite needs some fixes (there are probably some related issues already) but I would prefer to address that in a separate PR after the tutorial, to be safe.

So for now, if you approve, I would merge this PR as is.

**Pull request merged**